velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -88,12 +88,51 @@ SubstraitVeloxExprConverter::toVeloxExpr(
       return std::make_shared<core::ConstantTypedExpr>(
           veloxType, variant::null(veloxType->kind()));
     }
+    case ::substrait::Expression_Literal::LiteralTypeCase::kString:
+      return std::make_shared<core::ConstantTypedExpr>(
+          variant(substraitLit.string()));
+    case ::substrait::Expression_Literal::LiteralTypeCase::kList: {
+      // Literals in List are put in a constant vector.
+      std::vector<::substrait::Expression::Literal> literals;
+      literals.reserve(substraitLit.list().values().size());
+      for (const auto& literal : substraitLit.list().values()) {
+        literals.emplace_back(literal);
+      }
+      return literalsToConstantExpr(literals);
+    }
     default:
       VELOX_NYI(
           "Substrait conversion not supported for type case '{}'", typeCase);
   }
 }
 
+std::shared_ptr<const core::ConstantTypedExpr>
+SubstraitVeloxExprConverter::literalsToConstantExpr(
+    const std::vector<::substrait::Expression::Literal>& literals) {
+  std::vector<variant> variants;
+  variants.reserve(literals.size());
+  VELOX_CHECK_GE(literals.size(), 0, "List should have at least one item.");
+  std::optional<TypePtr> literalType = std::nullopt;
+  for (const auto& literal : literals) {
+    auto veloxVariant = toVeloxExpr(literal)->value();
```

Comment:
Does this work for nested arrays, i.e. array(array(bigint))?

Reply:
yes it works.  I add a new UT case for nested arrays.